android/src/main/java/com/revenuecat/purchases/hybridcommon/common.kt

```diff
@@ -443,6 +476,21 @@ fun getPromotionalOffer() : ErrorContainer {
 
 // region private functions
 
+internal class InvalidProrationMode(): Exception()
+
+@Throws(InvalidProrationMode::class)
+internal inline fun getGoogleProrationMode(oldProductId: String?, prorationMode: Int?) : GoogleProrationMode?  {
+    return prorationMode
+        ?.takeIf { oldProductId != null }
+        ?.let { index ->
+            GoogleProrationMode.values().find {
+                it.playBillingClientMode == prorationMode
+            } ?: run {
+                throw InvalidProrationMode()
+            }
+        }
+}
+
```

Comment:
I wrote this so many different ways with lambdas, `Pair`s, copy/pasted implementation but this one seemed to be the least worst 🤷‍♂️ 

- Allows for returning of a nullable `GoogleProrationMode` without it being an error
- Testable code
- Clear error when not a valid value

Reply:
Hmm we do have the [Result](https://github.com/RevenueCat/purchases-android/blob/main/utils/src/main/java/com/revenuecat/purchases/utils/Result.kt) class in android that could fit here... We can't use it directly here (since it's internal) so we would have to duplicate it and might remove some duplication.

But I think this is probably ok.

Reply:
I did look into using Result as well 😅 I pretty much hated how all of the solutions to this looked 😛